realm/src/androidTest/java/io/realm/internal/JNITransactions.java

```diff
@@ -376,4 +378,13 @@ public void testAddEmptyRowWithPrimaryKeyLong() {
         assertEquals(1, t.size());
         assertEquals(42, t.getRow(rowIndex).getLong(0));
     }
+
+    public void testPrimaryKeyTableMigration() throws IOException {
```

Comment:
Just being able to ask for the primary key without crashing actually tests most of the logic, but you are right that testing the primary key hasn't changed wasn't among them. It has been added.